File: beginner/python/1827.py
def matrix(tamanho):
    m = [[0 for _ in range(tamanho)] for _ in range(tamanho)]

    for linha in range(tamanho):
        for coluna in range(tamanho):
            if linha == coluna:
                m[linha][coluna] = 2
            elif linha == tamanho - coluna - 1:
                m[linha][coluna] = 3
            else:
                m[linha][coluna] = 0

    inicio = tamanho // 3
    fim = tamanho - inicio

    for linha in range(inicio, fim):
        for coluna in range(inicio, fim):
            m[linha][coluna] = 1

    meio = tamanho // 2
    m[meio][meio] = 4

    for linha in range(tamanho):
        for coluna in range(tamanho):
            print(m[linha][coluna], end="")
        print()
    
    print()

def main():
    while True:
        try:
            tamanho = int(input())
            matrix(tamanho)
        except EOFError:
            break

if __name__ == "__main__":
    main()

File: beginner/javascript/1827.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function matrix(tamanho) {
  const m = Array.from({ length: tamanho }, () => Array(tamanho).fill(0));

  for (let linha = 0; linha < tamanho; linha++) {
    for (let coluna = 0; coluna < tamanho; coluna++) {
      if (linha === coluna) {
        m[linha][coluna] = 2;
      } else if (linha === tamanho - coluna - 1) {
        m[linha][coluna] = 3;
      } else {
        m[linha][coluna] = 0;
      }
    }
  }

  const inicio = Math.floor(tamanho / 3);
  const fim = tamanho - inicio;

  for (let linha = inicio; linha < fim; linha++) {
    for (let coluna = inicio; coluna < fim; coluna++) {
      m[linha][coluna] = 1;
    }
  }

  const meio = Math.floor(tamanho / 2);
  m[meio][meio] = 4;

  for (let linha = 0; linha < tamanho; linha++) {
    console.log(m[linha].join(""));
  }

  console.log();
}

rl.on('line', (input) => {
  const tamanho = parseInt(input);
  matrix(tamanho);
});

